src/tensors/tensor.jl

```diff
@@ -3,152 +3,114 @@
-             {T,S<:IndexSpace,N₁,N₂,A<:DenseMatrix{T}}
+    function TensorMap{T,S,N₁,N₂,A}(data::A, space::TensorMapSpace{S,N₁,N₂}) where {T,S<:IndexSpace,N₁,N₂,A<:DenseVector{T}}
         T ⊆ field(S) || @warn("scalartype(data) = $T ⊈ $(field(S)))", maxlog = 1)
-        return new{T,S,N₁,N₂,Trivial,A,Nothing,Nothing}(data, codom, dom)
+        return new{T,S,N₁,N₂,A}(data, space)
     end
 end
 #! format: on
 
 """
-    Tensor{T, S, N, I, A, F₁, F₂} = TensorMap{T, S, N, 0, I, A, F₁, F₂}
+    Tensor{T, S, N, A<:DenseVector{T}} = TensorMap{T, S, N, 0, A}
 
 Specific subtype of [`AbstractTensor`](@ref) for representing tensors whose data is stored
-in blocks of some subtype of `DenseMatrix`.
+in a dense vector.
 
-A `Tensor{T, S, N, I, A, F₁, F₂}` is actually a special case `TensorMap{T, S, N, 0, I, A, F₁, F₂}`,
+A `Tensor{T, S, N, A}` is actually a special case `TensorMap{T, S, N, 0, A}`,
 i.e. a tensor map with only a non-trivial output space.
 """
-const Tensor{T,S,N,I,A,F₁,F₂} = TensorMap{T,S,N,0,I,A,F₁,F₂}
-
-"""
-    TrivialTensorMap{T, S, N₁, N₂, A<:DenseMatrix} = TensorMap{T, S, N₁, N₂, Trivial, 
-                                                                        A, Nothing, Nothing}
-
-A special case of [`TensorMap`](@ref) for representing tensor maps with trivial symmetry,
-i.e., whose `sectortype` is `Trivial`.
-"""
-const TrivialTensorMap{T,S,N₁,N₂,A<:DenseMatrix} = TensorMap{T,S,N₁,N₂,Trivial,A,Nothing,
-                                                             Nothing}
-
-"""
-    TrivialTensor{T, S, N, A} = TrivialTensorMap{T, S, N, 0, A}
-
-A special case of [`Tensor`](@ref) for representing tensors with trivial symmetry, i.e.,
-whose `sectortype` is `Trivial`.
-"""
-const TrivialTensor{T,S,N,A} = TrivialTensorMap{T,S,N,0,A}
-
-# TODO: check if argument order should change
-"""
-    tensormaptype(::Type{S}, N₁::Int, N₂::Int, [::Type{T}]) where {S<:IndexSpace,T} -> ::Type{<:TensorMap}
+const Tensor{T,S,N,A} = TensorMap{T,S,N,0,A}
 
-Return the fully specified type of a tensor map with elementary space `S`, `N₁` output
-spaces and `N₂` input spaces, either with scalar type `T` or with storage type `T`.
-"""
-function tensormaptype(::Type{S}, N₁::Int, N₂::Int,
-                       ::Type{TorA}) where {S,TorA<:MatOrNumber}
-    I = sectortype(S)
-    if TorA <: DenseMatrix
-        M = TorA
-        T = scalartype(TorA)
-    elseif TorA <: Number
-        M = Matrix{TorA}
-        T = TorA
-    else
-        throw(ArgumentError("the final argument of `tensormaptype` should either be the scalar or the storage type, i.e. a subtype of `Number` or of `DenseMatrix`"))
-    end
-    if I === Trivial
-        return TensorMap{T,S,N₁,N₂,I,M,Nothing,Nothing}
+# TODO: should we deprecate this?
+# It seems quite heavily used in MPSKit.jl, although they are all of the form
+# `tensormaptype(S, N₁, N₂, A)` and so could thus be replaced with `TensorMap{S, N₁, N₂, A}`
```

Comment:
We could, but it should then be `TensorMap{scalartype(A),S,N1,N2,A}`. I have no strong opinion about this, but I quite like that I don't have to think about `storagetype` vs `scalartype`, so it does serve a purpose.

Reply:
ok, fair enough, I keep it in and can probably use it at other places.